thorvg/CMakeLists.txt

```diff
@@ -0,0 +1,109 @@
+# Set a name for libthorvg library
+set(TVG_LIB libthorvg)
+set(TVG_INC_DIR thorvg/src/bindings/capi)
+set(TVG_SUBDIR "${CMAKE_CURRENT_SOURCE_DIR}/thorvg")
+set(TVG_BUILD_DIR "${CMAKE_BINARY_DIR}/thorvg_build")
+set(TVG_CROSS_CFG "${CMAKE_BINARY_DIR}/thorvg_build/cross_file.txt")
+idf_build_get_property(python PYTHON)
+
+idf_component_register(SRCS dummy.c
+                    INCLUDE_DIRS "${TVG_INC_DIR}")
+
+include(ExternalProject)
+
+if(NOT CMAKE_CXX_FLAGS STREQUAL "")
+    string(REGEX MATCHALL "-[^ ]+" match_list "${CMAKE_CXX_FLAGS}")
+    set(MESON_CXX_FLAGS "")
+
+    list(LENGTH match_list match_list_length)
+    math(EXPR last_index "${match_list_length}-1")
+
+    foreach(index RANGE ${last_index})
+        list(GET match_list ${index} match)
+        string(APPEND MESON_CXX_FLAGS "'${match}'")
+        if(NOT ${index} EQUAL ${last_index})
+            string(APPEND MESON_CXX_FLAGS ", ")
+        endif()
+    endforeach()
+else()
+    string(APPEND MESON_CXX_FLAGS "''")
+endif()
+
+if(NOT CMAKE_EXE_LINKER_FLAGS STREQUAL "")
+    string(REGEX MATCHALL "-[^ ]+" match_list "${CMAKE_EXE_LINKER_FLAGS}")
+    set(MESON_LINKER_FLAGS "")
+
+    list(LENGTH match_list match_list_length)
+    math(EXPR last_index "${match_list_length}-1")
+
+    foreach(index RANGE ${last_index})
+        list(GET match_list ${index} match)
+        string(APPEND MESON_LINKER_FLAGS "'${match}'")
+        if(NOT ${index} EQUAL ${last_index})
+            string(APPEND MESON_LINKER_FLAGS ", ")
+        endif()
+    endforeach()
+else()
+    string(APPEND MESON_LINKER_FLAGS "''")
+endif()
+
+string(LENGTH "${CMAKE_CXX_COMPILER}" path_length)
+math(EXPR extract_length "${path_length} - 3")
+string(SUBSTRING "${CMAKE_CXX_COMPILER}" 0 ${extract_length} TOOLCHAIN_PREFIX)
+
+file(WRITE ${TVG_CROSS_CFG} "
+[binaries]
+cpp = '${TOOLCHAIN_PREFIX}g++'
+ar = '${TOOLCHAIN_PREFIX}ar'
+as = '${TOOLCHAIN_PREFIX}as'
+ranlib = '${TOOLCHAIN_PREFIX}ranlib'
+ld = '${TOOLCHAIN_PREFIX}ld'
+strip = '${TOOLCHAIN_PREFIX}strip'
+
+[built-in options]
+cpp_args = ['-D_GNU_SOURCE','-D__linux__',${MESON_CXX_FLAGS}]
+")
+
+if(NOT MESON_LINKER_FLAGS STREQUAL "''")
+    file(APPEND ${TVG_CROSS_CFG} "
+cpp_link_args = [${MESON_LINKER_FLAGS}]
+")
+endif()
+
+file(APPEND ${TVG_CROSS_CFG} "
+[host_machine]
+system = 'android'
```

Comment:
Interesting, why set the host to android?

Reply:
> Interesting, why set the host to android?

```
if get_option('threads') == true and host_machine.system() != 'windows' and host_machine.system() != 'android'
thread_dep = meson.get_compiler('cpp').find_library('pthread')
thorvg_lib_dep += [thread_dep]
endif
```

To solve the `thread_dep` error, but later added `-Dthreads=false`, I think it is better to write `freertos` here now.

Reply:
We do provide pthreads library, so enabling thread support should work fine. What was the exact error you got here?

Reply:
> We do provide pthreads library, so enabling thread support should work fine. What was the exact error you got here?

```
Project name: thorvg
Project version: 0.13.6
C++ compiler for the host machine: /home/xuxin/.espressif/tools/riscv32-esp-elf/esp-13.2.0_20240530/riscv32-esp-elf/bin/riscv32-esp-elf-g++ (gcc 13.2.0 "riscv32-esp-elf-g++ (crosstool-NG esp-13.2.0_20240530) 13.2.0")
C++ linker for the host machine: /home/xuxin/.espressif/tools/riscv32-esp-elf/esp-13.2.0_20240530/riscv32-esp-elf/bin/riscv32-esp-elf-g++ ld.bfd 13.2.0
C++ compiler for the build machine: ccache c++ (gcc 11.4.0 "c++ (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0")
C++ linker for the build machine: c++ ld.bfd 2.38
Build machine cpu family: x86_64
Build machine cpu: x86_64
Host machine cpu family: Generic
Host machine cpu: esp
Target machine cpu family: Generic
Target machine cpu: esp
Configuring config.h using configuration

../../../../thorvg/src/meson.build:47:43: ERROR: C++ shared or static library 'pthread' not found

A full log can be found at /home/xuxin/esp_work/idf-extra-components_fork/thorvg/examples/thorvg-example/build/thorvg_build/meson-logs/meson-log.txt
```

Reply:
I see, this should be possible to solve by adding pthread component library to the linker flags.

We can keep this as a follow-up, I think it's okay to not fix this in the current MR. (Provided that thorvg is used from one FreeRTOS task only.)